### ISSUE TITLE
Paredit interaction with strings

### ISSUE THREAD

**Issue by Cyrik** [BetterThanTomorrow/calva#1440]
When I do “slurp sexp forward” at:

`"|"somestuff`

where the | is my cursor, I get ” somestuff”.

I would expect the string to not start with a space.
I've tried a few other paredit commands inside of strings and they seem Ok, but I'm to a paredit expert.

According to @PEZ :

> Not really intended, It’s a side effect of that strings are treated as lists and we haven’t made a special case for it. But we can consider it.

**Comment by PEZ** [comment]
Thanks. To clarify the strings-as-lists a bit. Since when you slurp into a list this space is cleaned away. That is because we run the formatter on the edited text, and it removes surrounding whitespace. Formatting with in a string does not do this.

We have a few more edits that behave a bit strangely around strings:

```clojure
foo "|"
```

<kbd>slurp backward</kbd> => 

```clojure
"foo |"
```

---

```cojure
f|oo ""
```

<kbd>drag forward down</kbd> =>

```clojure
"f|oo "
```

---

```cojure
"" f|oo
```

<kbd>drag backward down</kbd> =>

```clojure
"
f|oo"
```

---

Maybe more, but these are the ”slurpy” ones I come to think of.

**Comment by Cyrik** [comment]
Yes, it looks like the problem is always pulling into strings where some sort of white space is introduced.

**Comment by jramosg** [comment]
Closed by https://github.com/BetterThanTomorrow/calva/pull/3005 and https://github.com/BetterThanTomorrow/calva/pull/3006
